Website/Providers/DataProviders/SqlDataProvider/08.00.01.01.SqlDataProvider

```diff
@@ -445,3 +445,913 @@ AS
 
         EXEC(@sql)
 GO
+
+/* DNN-8378 Start */
+/* Create HasBeenPublished field */
+IF NOT EXISTS
+(SELECT * FROM sys.all_columns c INNER JOIN sys.tables t
+	ON c.object_id = t.object_id AND t.name = '{objectQualifier}Files'
+	WHERE c.name = 'HasBeenPublished')
+BEGIN
+	ALTER TABLE {databaseOwner}[{objectQualifier}Files]
+		ADD HasBeenPublished bit DEFAULT 1 NOT NULL		
```

Comment:
the Flag "HasBeenPublished" is not really self-explaning and not sufficient. 
 If you continue the life-cycle of a file on a website, there will be a time when a cleanup is planned and a cleanup workflow identifies files to be deleted, i.e. another flag would be needed "readyForDelete" or "Deleted".
 Instead of a boolean flag I suggest a "PublishState" (tinyInt), where the different states can be stored (and a master table FileStates, where states are explained).